### PR TITLE
improve documentation

### PR DESCRIPTION
**Pull request by RaphaelVogel**

### CODE REVIEW

File: concourse/model/traits/notifications.py [hunk 1]
```diff
@@ -32,16 +33,39 @@
 )
 
 
-class NotificationTriggeringPolicy(enum.Enum):
+class NotificationTriggeringPolicy(AttribSpecMixin, enum.Enum):
```
Comment (ccwienk) [inline concourse/model/traits/notifications.py]: as just discussed via slack: extending this enum seems like a clever idea. my only objection would be that - different to all other attrib-specs we so far have - the names / attributes here do not refer to actual sub-attributes, but to values of one common attribute. How does the rendered documentation page look like?

File: doc/lib/attributes.py [hunk 2]
```diff
@@ -76,6 +77,8 @@ def attr_to_table_row(attr_spec, prefix=None):
                 # recurse to child element
                 if issubclass(type_, model.NamedModelElement):
                     child_element = type_(name='<user-chosen>', raw_dict={})
```
Comment (ccwienk) [inline doc/lib/attributes.py]: as discussed yesterday: there should be no need to instantiate the elements. just use `type_` in all cases

File: concourse/model/traits/notifications.py (hunk 1 above)
Comment (RaphaelVogel) [reply concourse/model/traits/notifications.py]: Changed the heading into `Attribute Values` in case of enum

File: doc/lib/attributes.py (hunk 2 above)
Comment (RaphaelVogel) [reply doc/lib/attributes.py]: removed instantiation of elements

**Pull request merged**